### PR TITLE
TST: Disable testing that hangs Azure CI

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

Precursor to https://github.com/pandas-dev/pandas/pull/45478

This PR doesn't necessarily guarantee that we won't see random Azure timeouts, but based on working on #45478, these were tests that noticeably caused the CI to hang

### CODE REVIEW

File: pandas/util/_test_decorators.py [hunk 1]
```diff
@@ -266,7 +266,8 @@ def file_leak_context():
     ContextManager analogue to check_file_leaks.
     """
     psutil = safe_import("psutil")
-    if not psutil:
+    if not psutil or is_platform_windows():
+        # Checking for file leaks can hang on Windows CI
```
Comment (jbrockmendel) [inline pandas/util/_test_decorators.py:270]: is there a reference for this claim?

File: pandas/tests/window/test_online.py [hunk 2]
```diff
@@ -9,6 +15,15 @@
 )
 import pandas._testing as tm
 
+# TODO: Mark these as pytest.mark.single GH 44584
```
Comment (jbrockmendel) [inline pandas/tests/window/test_online.py:18]: NBD for future reference: i've been using the pattern "TODO(GH#12345): foo-bar" for TODOs with particular issues that will resolve them

File: pandas/util/_test_decorators.py (hunk 1 above)
Comment (mroeschke) [reply pandas/util/_test_decorators.py:270]: Just anecdotally while investigating https://github.com/pandas-dev/pandas/pull/45478.

I noticed disabling the analog check in the excel tests would no longer hang the build around the excel files

**Comment by jreback** [comment]
@mroeschke backport?

**Comment by mroeschke** [comment]
> @mroeschke backport?

Sure might be helpful for the 1.4.x branch as well.

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**